### DISCUSSION TITLE
How can I hide certain column from Add row page ?

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo_admin#286] [Q&A]
How can I hide /remove any column from add row  ( Let's say Email column considering below image) and how can I assign it's value from the code .
![image](https://user-images.githubusercontent.com/45959281/217194020-139eacc9-a252-4d58-a3c8-6f1c99d51234.png)

I have a `updated_by` field and I want that updated_by will be the current  user  updating the data. So, it is not good that `updated_by` field will be visible to the person updating the data. And also `updating_by` field should be assigned by the `user_id` updating the data.

**Comment by dantownsend** [comment]
So this is for custom forms?

Rather than putting `updated_by` in the form, you can access the user from the request.

```python
class MyModel(pydantic.BaseModel):
    message: str = "hello world"


def my_endpoint(request: Request, data: MyModel):
    # to get the logged in user:
    user = request.user.user
    # do stuff
    return "Successful."


config = FormConfig(
    name="My Form",
    pydantic_model=MyModel,
    endpoint=my_endpoint
)
```

**Reply by sumitsharansatsangi** [reply]
No, not for the custom forms. I think the picture attached mislead you. It's for the adding the table, The form I mean is the form comes after pressing `Add row ` button.

**Reply by dantownsend** [reply]
Ah, OK. We don't have anything like that at the moment. I'd like to add `read_only_columns` to `TableConfig`, and those just have the default values and can't be modified by the user.

**Reply by sumitsharansatsangi** [reply]
Please do it . Thank you.

**Reply by sumitsharansatsangi** [reply]
@dantownsend , @sinisaos , Can you please look into the `read_only_columns` option for its implementation?

**Reply by sinisaos** [reply]
@sumitsharansatsangi PR [is here](https://github.com/piccolo-orm/piccolo_admin/pull/289) and everything works, but Daniel has to review it, and if it's good enough it will be merged.

**Reply by sumitsharansatsangi** [reply]
Thanks @sinisaos , for the PR. @dantownsend , Can you please review it?
